// for now use counters as in openconfig demo

syntax = "proto3";
package counters;


// RPC to get map id and name on interface
service int_map{
    rpc GetMaps(MappacketRequest) returns (MappaketReply) {};
}
// RPC to update map 
service int_UpdateMap{
    rpc UpdateMap(UpdateMapRequest) returns (UpdateMapReply) {};
}


// packet request for GetMaps
message MappacketRequest{
    string interface = 1;
}

// packet reply for GetMaps
message MappaketReply{
    repeated MapInfo map_info = 1;
}
// message to be repeated in MappacketReply
message MapInfo{
     int32  id = 1;
     string name  = 2;
     repeated MapEntry entries = 3; 
}

// packet for UpdateMap request , map_id , subnet as x.x.x.x/24 , lbl as int
message UpdateMapRequest{
    int32 map_id = 1;
    string subnet = 2;
    int32 lbl = 3;
}
// packet for UpdateMap reply , string for now with confirmation if it was programmed or not
message UpdateMapReply{
     string message = 1;
}

message MapEntry{
  int32 subnet = 1;
  string ipaddress = 2;
  int32 lbl = 3 ;
}
